### ISSUE TITLE
Omnibar: Remote data source?

### ISSUE THREAD

**Issue by jackjwilliams** [palantir/blueprint#2528]
#### Context

So far I've only seen the ability (on any of the "Select" type controls, including Omnibar), to pass in a list of options. Is it possible to call out to an external datasource for querying? 

#### Examples

I have a Geo Dataset of about 29,000 searchable items. The user would open the omnibar, and begin typing to start querying the API. The API would return possible items for selection.

**Comment by giladgray** [comment]
@jackjwilliams yes it is entirely possible _in your app code_ to do this. i refuse to implement explicit async support in these components; instead we will eventually provide a single `loading` prop that you can enable while your app makes async calls (howsoever you would like to implement that).

**Comment by subratawebops** [comment]
@giladgray Can you provide an example to implement this?

**Comment by giladgray** [comment]
@subratawebops there are many great examples floating around on the webs detailing how to do async data loading in a React component. have a google!

**Comment by dmackerman** [comment]
> I think the remaining question here is how you would display the spinner to indicate a loading operation.

Maybe use the `emptyState` to display a Spinner?

**Comment by giladgray** [comment]
@dmackerman i would like to add a `loading` prop to all the components to visualize a loading state. it will be up to the app to do the actual async operation. let's continue on a new issue plese.

**Comment by danilofes** [comment]
> @dmackerman i would like to add a `loading` prop to all the components to visualize a loading state. it will be up to the app to do the actual async operation. let's continue on a new issue please.

@giladgray Is such issue already open? I think the `loading` prop is very much needed.
